examples/pi-home-router/src/router.rs

```diff
@@ -1,11 +1,27 @@
-use route_rs_runtime::link::{Link, LinkBuilder, PacketStream};
+use crate::arp::{ArpGenerator, ArpHandler};
+use route_rs_packets::MacAddr;
+use route_rs_runtime::link::primitive::ProcessLink;
+use route_rs_runtime::link::{Link, LinkBuilder, PacketStream, ProcessLinkBuilder};
+use route_rs_runtime::utils::test::packet_generators::immediate_stream;
+use std::collections::HashMap;
+use std::net::Ipv4Addr;
+use std::sync::{Arc, Mutex};
 
 /// Top level structure, implements LinkBuilder so it can be run by the test harness
 #[derive(Default)]
 pub(crate) struct Router {
     host: Option<String>,
     lan: Option<String>,
     wan: Option<String>,
+
+    // TODO: investigate/discuss concurrency options
+    // https://doc.rust-lang.org/std/sync/struct.RwLock.html
+    // https://docs.rs/chashmap/2.2.2/chashmap/
+    // https://github.com/jonhoo/rust-evmap
+    // https://github.com/jonhoo/flurry
+    /// Top-level ARP table reference.
+    /// Links that require a reference should clone this Arc.
+    arp_table: Arc<Mutex<HashMap<Ipv4Addr, MacAddr>>>,
```

Comment:
Another option https://github.com/xacrimon/dashmap.  Its concurrent out of the box, so perhaps this will be easier to use.